mythril/analysis/solver.py

```diff
@@ -109,29 +108,49 @@ def get_transaction_sequence(global_state, constraints):
             minimize.append(transaction.call_data.calldatasize)
             minimize.append(transaction.call_value)
 
-            concrete_transactions[tx_id] = tx_template.copy()
-
         else:
             creation_tx_ids.append(tx_id)
 
     model = get_model(tx_constraints, minimize=minimize)
-
+    min_price_dict = {}  # type: Dict[str, int]
     for transaction in transactions:
         tx_id = str(transaction.id)
-
-        concrete_transactions[tx_id]["calldata"] = "0x" + "".join(
+        concrete_transaction = dict()  # type: Dict[str, str]
+        concrete_transaction["input"] = "0x" + "".join(
             [
                 hex(b)[2:] if len(hex(b)) % 2 == 0 else "0" + hex(b)[2:]
                 for b in transaction.call_data.concrete(model)
             ]
         )
-
-        concrete_transactions[tx_id]["call_value"] = (
-            "0x%x"
-            % model.eval(transaction.call_value.raw, model_completion=True).as_long()
-        )
-        concrete_transactions[tx_id]["caller"] = "0x" + (
+        value = model.eval(transaction.call_value.raw, model_completion=True).as_long()
+        concrete_transaction["value"] = "0x%x" % value
+        caller = "0x" + (
             "%x" % model.eval(transaction.caller.raw, model_completion=True).as_long()
         ).zfill(40)
 
-    return concrete_transactions
+        concrete_transaction["origin"] = caller
+        concrete_transaction["address"] = "%s" % transaction.callee_account.address
+        concrete_transactions.append(concrete_transaction)
+        min_price_dict[caller] = min_price_dict.get(caller, 0) + value
+    if len(transactions) > 0:
+        initial_state = transactions[0].world_state.accounts
+    else:
+        for transaction in transaction_sequence:
+            if transaction.id == "1":
+                initial_state = transaction.world_state.accounts
+                break
+
+    states = {}
+    for address, account in initial_state.items():
```

Comment:
initial_state can be uninitialized

Reply:
There will always be a tx_id=“1” right?

Reply:
Right now, maybe. But we never document/test that this is always the case, so any refactor can change this by accident

Reply:
Now it should be fine